sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceImpl.java

```diff
@@ -67,7 +68,8 @@ public boolean equals(Object o) {
 				&& date.equals(other.getDate())
 				&& duration == other.getDuration()
 				&& getCount() == other.getCount()
-				&& lastVisitStartTime == other.getLastVisitStartTime();
+				&& lastVisitStartTime == other.getLastVisitStartTime()
+				&& currentOpenSessions == other.getCurrentOpenSessions();
```

Comment:
_you’re checking whether both references point to the same object in memory, not whether the two Date objects represent the same point in time. To compare the values of the two Date objects, you should use the .equals() method:_

Reply:
Good catch, it was like that for 15 years though.... Let's fix it now. And @st-manu , if you apply the changes from @ern changing `int` to `Integer`, currentOpenSessions should be compared using eqauls as well.